packages/wdio-cli/src/launcher.js

```diff
@@ -55,49 +55,56 @@ class Launcher {
 
     /**
      * run sequence
-     * @return  {Promise} that only gets resolves with either an exitCode or an error
+     * @param   {Function}  callback    callback
+     * @return  {Promise}               that only gets resolves with either an exitCode or an error
      */
-    async run () {
```

Comment:
- run function now accepts callback as argument and calls it in the end;
- moved `exitHook(:: this.exitHandler)` to the top so that I can emit `shutdown` event in callback
- no other changes, just wrapped everything with try/catch block